app/controllers/businessmatching/MsbSubSectorsController.scala

```diff
@@ -59,59 +61,14 @@ class MsbSubSectorsController @Inject()(val authConnector: AuthConnector,
             Future.successful(BadRequest(views.html.businessmatching.services(f, edit)))
 
           case ValidForm(_, data) =>
-            lazy val updateModel = for {
-              bm <- businessMatchingService.getModel
-              cache <- businessMatchingService.updateModel(if (data.msbServices.contains(TransmittingMoney)) {
-                bm.msbServices(Some(data))
-              } else {
-                bm.msbServices(Some(data)).clearPSRNumber
-              })
-              _ <- OptionT.liftF(updateMsb(bm.msbServices, data.msbServices, cache))
-            } yield cache
-
-            lazy val redirectResult = OptionT.some[Future, Result](if (data.msbServices.contains(TransmittingMoney)) {
-              Redirect(routes.PSRNumberController.get(edit))
-            } else {
-              Redirect(routes.SummaryController.get())
-            })
-
-            updateModel flatMap { _ => redirectResult } getOrElse InternalServerError("Could not update services")
+            dataCacheConnector.update[ChangeSubSectorFlowModel](ChangeSubSectorFlowModel.key) {
+              _.getOrElse(ChangeSubSectorFlowModel()).copy(subSectors = Some(data.msbServices))
+            } flatMap {
+              case Some(m@ChangeSubSectorFlowModel(Some(set), _)) if !(set contains TransmittingMoney) =>
```

Comment:
Not sure why you don't update the sector information for transmitting money

Reply:
if you have transmitting money then you don't do the save until submission of the psr number page